mlir/lib/Dialect/MIOpen/utilities.cpp

```diff
@@ -0,0 +1,66 @@
+//===- utilities.cpp - MIOpen utility functions ---------------------===//
```

Comment:
Nit: I think we can move this inside `GridwiseGemmToBLockwise.cpp` considering that this has been only used there.

This catch my attention because of the file names as `utilities.cpp` or `common.h` sounds like a bad idea. I'd also be fine with you keep this compilation as-is but rename the header/cpp.

Reply:
I could probably pull this up a level into `Dialect/MIOpen/utilities.h`

This particular change came out of the test pattern refactor, where I wanted `createConstantFloatOp()` in `miopen-gen`. Since I was restructuring everything, I thought I'd stick it here.

Reply:
(also, the functions are used in a few places in lowering)

My general sense is that having a file into which we can chuck utility routines that don't obviously belong anywhere else yet is a good call?

Reply:
Yes, avoiding duplication is always a good thing. I'm trying to be defensive on a slightly different topic: If a component doesn't have a specific enough purpose it can evolve into something completely unexpected (I guess LowerMIOpenOps.h is one such example). To prevent such things from happening giving it a specific enough purpose/naming might ring a bell to developer/reviewers.

Reply:
Hm ... yeah, something like `constantUtils.cpp` or `builderUtils.cpp` might be a good bit of renaming - it leaves the file generic enough that it might collect some other widely-used patterns, and it'd at least match things like `loweringUtils`